presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoInstaller.java

```diff
@@ -85,21 +95,129 @@ private static void installUsingRpm(String pathToRpm, boolean checkDependencies)
+        }
+        createDir(TAR_INSTALLATION_DIR.resolve("etc"));
+        addTpchCatalog(TAR_INSTALLATION_DIR.resolve("etc/catalog"));
+    }
+
+    private static void preInstallChecks()
+            throws PrestoManagerException
+    {
+        if (Float.valueOf("1.8").compareTo(Float.valueOf(System.getProperty("java.specification.version"))) < 0) {
+            throw new PrestoManagerException("Java version 8 or above is required");
+        }
+    }
+
+    public static void addConfigFiles()
+            throws PrestoManagerException
+    {
+        addConfigProperties();
+        addEnvSh();
+        addJvmConfig();
+        addNodeProperties();
+    }
+
+    private static void addConfigProperties()
+            throws PrestoManagerException
+    {
+        Properties properties = new Properties();
+        properties.setProperty("coordinator", "true");
+        properties.setProperty("node-scheduler.include-coordinator", "true");
+        properties.setProperty("http-server.http.port", "8080");
+        properties.setProperty("discovery-server.enabled", "true");
+        properties.setProperty("discovery.uri", "http://localhost:8080");
+        try (OutputStream output = newOutputStream(TAR_INSTALLATION_DIR.resolve("etc/config.properties"), CREATE_NEW)) {
+            properties.store(output, "single node install config");
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to add config.properties", e);
+        }
+    }
+
+    private static void addNodeProperties()
+            throws PrestoManagerException
+    {
+        // TODO: Add node.server-log-file & node.launcher-log-file
+        Properties properties = new Properties();
+        properties.setProperty("node.environment", "test");
+        properties.setProperty("node.id", randomUUID().toString());
+        properties.setProperty("node.data-dir", TAR_DATA_DIR.toAbsolutePath().toString());
+        properties.setProperty("plugin.config-dir", TAR_INSTALLATION_DIR.resolve("etc/catalog").toString());
+        properties.setProperty("plugin.dir", TAR_INSTALLATION_DIR.resolve("plugin").toString());
+        try (OutputStream output = newOutputStream(TAR_INSTALLATION_DIR.resolve("etc/node.properties"), CREATE_NEW)) {
+            properties.store(output, "configurations specific to the node");
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to add config.properties", e);
+        }
+    }
+
+    private static void addJvmConfig()
+            throws PrestoManagerException
+    {
+        String jvmConfig = "-server\n" +
+                "-Xmx16G\n" +
+                "-XX:-UseBiasedLocking\n" +
+                "-XX:+UseG1GC\n" +
+                "-XX:+ExplicitGCInvokesConcurrent\n" +
+                "-XX:+HeapDumpOnOutOfMemoryError\n" +
+                "-XX:+UseGCOverheadLimit\n" +
+                "-XX:+ExitOnOutOfMemoryError\n" +
+                "-XX:ReservedCodeCacheSize=512M";
```

Comment:
We should not be hard-coding this sort of thing.

If we want there to be _any_ default configuration for Presto, we should include that in the configuration for Presto Manager. Better yet, we can provide default Presto configuration by mirroring the Presto configuration directory in one of Presto Manager's directories, and coping its contents after installation.